config_splunk.go

```diff
@@ -0,0 +1,66 @@
+package evergreen
+
+import (
+	"github.com/mongodb/grip/send"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type SplunkConfig struct {
+	SplunkConnectionInfo send.SplunkConnectionInfo `bson:",inline" json:",inline" yaml:",inline"`
+}
+
+func (c *SplunkConfig) SectionId() string { return "splunk" }
+
+func (c *SplunkConfig) Get(env Environment) error {
+	ctx, cancel := env.Context()
+	defer cancel()
+	coll := env.DB().Collection(ConfigCollection)
+
+	res := coll.FindOne(ctx, byId(c.SectionId()))
+	if err := res.Err(); err != nil {
+		if err == mongo.ErrNoDocuments {
```

Comment:
It might read nicer to say `if err != mongo.ErrNoDocuments { return err }` so that the rest of the code can not be so nested.